include/swift/AST/Module.h

```diff
@@ -378,6 +401,12 @@ class ModuleDecl : public DeclContext, public TypeDecl {
                           VisibleDeclConsumer &Consumer,
                           NLKind LookupKind) const;
 
+  /// This is a hack for 'main' file parsing and the integrated REPL.
```

Comment:
Because even though the REPL parses declarations in their own source file before adding them to the “combined” file, we still appear to do name lookups in the combined file, so we need a way to clear the cache.

Reply:
I’m actually still debugging some of this cache invalidation stuff. Ideally it would not be needed but we do lots of tricky things still.